### PR TITLE
Add prose, JS/TS & Rust guidelines used by Unboxed and prof-education

### PR DESCRIPTION
**Pull request by mikemaccana**
There's an existing [CONTRIBUTING.md](https://github.com/Unboxed-Software/solana-course/blob/main/CONTRIBUTING.md) used by the Unboxed Course, created by James and I, and used for both the Unboxed Course and Professional Education programs. Per Slack, ideally we could delete 90% of it and just refer to the one in this repo, with some added bits.

### CODE REVIEW

**Comment by mikemaccana** [comment]
I've merged https://github.com/solana-foundation/developer-content/pull/199 into this one and made the following changes:

 - Recommend to use the grammarly app for checking English instead of a vscode extension?
 - Moved above content guidelines 

I've also added a suggestion of adding Solana words to the Grammarly dictionary.

**Comment by mikemaccana** [comment]
@nickfrosty linted. Don't know why VScode auto-save doesn't work on this repo but oh well.

**Comment by mikemaccana** [comment]
@nickfrosty can you please check this out?

**Comment by jacobcreech** [comment]
@mikemaccana I'm not sure if it makes sense to copy over the contributing guide from unboxed to this repo. They're fundamentally different repositories.

Our time is better spent on other things since we already have a decent contributing guide.

**Comment by nickfrosty** [comment]
Yeah, I agree with Jacob

**Comment by mikemaccana** [comment]
@jacobcreech I absolutely disagree that `developer-content`, `professional-education`, and `solana-course` are fundamentally different - please check the contents on this PR and judge for yourself. There's valuable content to ensure the quality and consistency of JS, Anchor, prose, etc., regardless of the purpose of the developer content, some of which has been needed by this repo (e.g., getting people into the habit of running Grammarly).

**Comment by mikemaccana** [comment]
@jacobcreech we also want to get this merged before the Superteam bounty, so the new updates are consistent with recent Unboxed work.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -65,16 +29,130 @@ forth here.
...
+
+#### JS/TS
+
+We're trying to focus on Solana, not teaching JS/TS development and setup. This
+means reducing the JS/TS concepts needed to understand our demo code.
+
+- `.ts` files are run with `esrun`, which supports top-level `await`, doesn't
+  require a `tsconfig.json`, etc. There is no need for `async function main()`
+  wrappers or [IIFEs](https://developer.mozilla.org/en-US/docs/Glossary/IIFE).
+  `await` just works. If you see these wrappers, remove them.
```
Comment (mikemaccana) [inline CONTRIBUTING.md:121]:
```suggestion
- On the command line (not in the browser),`.ts` files are run on the command line with `esrun`, which supports top-level `await`, doesn't require a `tsconfig.json`, etc. There is no need for `async function main()` wrappers or [IIFEs](https://developer.mozilla.org/en-US/docs/Glossary/IIFE). `await` just works. If you see these wrappers, remove them.
```

Comment (mikemaccana) [reply CONTRIBUTING.md:121]: I spoke with @jacobcreech yesterday and wanted to make it more explicit that this is for ..ts` files on the command line. React sadly has a whole bunch of functions that aren't async and still need IIFEs to get an async context and use `await`.

**Review by mikemaccana — COMMENTED** [review]
Just tweaking my own PR, see conversation.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -65,16 +29,127 @@ forth here.
...
-### Prettier
-
-This repo uses prettier to help maintain the quality and consistency of the
-content within. There is a master
-[prettier configuration file](https://github.com/solana-foundation/developer-content/blob/main/.prettierrc)
-(`.prettierrc`) in the root of this repo.
+### Prose
+
+The guidelines below are consistent with Solana Foundation Style (if you work at
+Solana Foundation, see Notion), and
```
Comment (jacobcreech) [inline CONTRIBUTING.md:35]: Seeing as this is outward-facing style guide, best to not reference any internal documentation.
```suggestion
The guidelines below are consistent with Solana Foundation Style Guide, and
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -65,16 +29,127 @@ forth here.
...
+  [instruction handlers](https://solana.com/docs/terminology#instruction-handler)
+  as instructions! The reason is simple: an instruction cannot process an
+  instruction. The `multiple` template in Anchor also calls the functions
+  `handler`.
+
+Finally, run Grammarly over your PRs. While your spelling might be fine,
+Grammarly will often catch dangling words and consistencies over your prose. You
+don't have to change everything Grammarly asks, but checking your document will
+save other people time. For VScode users, there is a
+[VScode extension for Grammarly](https://marketplace.visualstudio.com/items?itemName=znck.grammarly).
```
Comment (jacobcreech) [inline CONTRIBUTING.md:91]: Paragraph is a duplicate of the first bullet point.
```suggestion
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -65,16 +29,127 @@ forth here.
...
+  have to work out that 'thg' means 'thing'.
+- Avoid repetitive, copy-paste code. This helps others change the code easily,
+  as they can update and fix things in a single place. If there's some
+  boilerplate, Solana-specific code you always need
+  [make a PR to the 'helpers' repository](https://github.com/solana-developers/helpers).
+- Avoid magic numbers. Nobody should see a `+ 32` in your code and wonder what
+  the `32` means. See Anchor notes below for Anchor-specific advice.
+- Avoid asking students to clone a git repo. The idea is that students should be
+  able to create projects from scratch when they have finished the course.
+  Referring to the code students have made in previous chapters is fine.
```
Comment (jacobcreech) [inline CONTRIBUTING.md:111]: This is not consistent across all guides. Imo remove

**Review by jacobcreech — APPROVED** [review]
Moving forward with this

**Pull request merged**